File: src/main/java/uss/code/course/dto/response/DepartmentResponse.java
package uss.code.course.dto.response;

import lombok.Builder;
import uss.code.member.domain.MemberDepartment;

import static lombok.AccessLevel.PRIVATE;

@Builder(access = PRIVATE)
public record DepartmentResponse(
        String code,

        String name
) {
    public static DepartmentResponse from(final MemberDepartment department) {
        return DepartmentResponse.builder()
                .code(department.name())
                .name(department.getName())
                .build();
    }
}

File: src/main/kotlin/uss/code/course/dto/response/DepartmentResponse.kt
package uss.code.course.dto.response

import uss.code.course.domain.CourseDepartment

data class DepartmentResponse(
    val code: String,

    val name: String,
) {
    companion object {
        fun from(department: CourseDepartment): DepartmentResponse {
            return DepartmentResponse(
                code = department.name,
                name = department.displayName,
            )
        }
    }
}
